app/src/main/java/com/example/concordia_campus_guide/Fragments/PathInfoCardFragment/PathInfoCardFragment.java

```diff
@@ -56,43 +53,45 @@ public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup c
 
         RecyclerView.LayoutManager layoutManager = new LinearLayoutManager(getContext());
         recyclerView.setLayoutManager(layoutManager);
-        directionList = new ArrayList<>();
-
-        Serializable temporaryDirectionsResults = getArguments().getSerializable("directionsResult");
-        directionsResults = (ArrayList<DirectionWrapper>) temporaryDirectionsResults;
-        walkingPointList = (ArrayList<WalkingPoint>) getArguments().getSerializable("walkingPoints");
-
+        updateDirectionResults();
         pathInfoCardViewModel = new ViewModelProvider(this).get(PathInfoCardViewModel.class);
-
-
         return view;
     }
 
+    private void updateDirectionResults() {
+        Serializable temporaryDirectionsResults = getArguments().getSerializable("directionsResult");
+        directionsResults = (ArrayList<PathInfoCard>) temporaryDirectionsResults;
+    }
+
     @Override
     public void onActivityCreated(@Nullable Bundle savedInstanceState) {
         super.onActivityCreated(savedInstanceState);
-        getIndoorOutdoorInfo();
-        setStartAndEndTime();
+        totalDuration = 0;
+        //wait sufficient time for the google outdoor direction api callback
+        (new Handler()).postDelayed(this::drawIndoorOutdoorInfo, 2500);
     }
 
-    public void getIndoorOutdoorInfo() {
-        LinearLayout layout = (LinearLayout) getView().findViewById(R.id.paths_image_buttons);
-        LinearLayout.LayoutParams layoutParams = new LinearLayout.LayoutParams(LinearLayout.LayoutParams.WRAP_CONTENT, LinearLayout.LayoutParams.WRAP_CONTENT, 1);
-        if (walkingPointList != null && directionsResults != null && !walkingPointList.isEmpty() && !directionsResults.isEmpty()) {
-            if (walkingPointList.get(0).getPointType() == PointType.ENTRANCE) {
-                populateIndoorDirectionsList();
-                setDividerTextView(layout, layoutParams);
-                populateOutdoorDirectionsList();
-            } else if (walkingPointList.get(walkingPointList.size() - 1).getPointType() == PointType.ENTRANCE) {
-                populateOutdoorDirectionsList();
-                setDividerTextView(layout, layoutParams);
-                populateIndoorDirectionsList();
+    public void drawIndoorOutdoorInfo() {
+        LinearLayout layout = getView().findViewById(R.id.paths_image_buttons);
+        LinearLayout.LayoutParams layoutParams = new LinearLayout.LayoutParams(LinearLayout.LayoutParams.WRAP_CONTENT, LinearLayout.LayoutParams.MATCH_PARENT, 1);
+        String type = "";
+        for (int i = 0; i < directionsResults.size(); i++) {
+            totalDuration += directionsResults.get(i).getDuration();
+            if(!directionsResults.get(i).getType().equals(type)){
+                type = directionsResults.get(i).getType();
+                if (i!=0) {
+                    setDividerTextView(layout, layoutParams);
+                }
+                makeIcon(layout, layoutParams, type);
             }
```

Comment:
if the i isnot necessary here, i would recommend using foreach instead to avoid any possibilities of arrayoutofboundry exception. Just a suggestion. :)

Reply:
yeah the if is necessary unfortunately :/

Reply:
it is ok. 👍